_includes/generator-form.html

```diff
@@ -0,0 +1,63 @@
+{::options parse_block_html="false" entity_output="as_input" /}
+<div class="tgmpa-generator-form">
+	<form id="generator-form" method="POST">
+		<input type="hidden" name="tgmpa_generate" value="1" />
+
+		<section class="tgmpa-form-primary">
+			<label for="tgmpa-flavor">I'm going to use TGMPA in a:<span class="required">*</span></label>
+			<fieldset>
+				<input type="radio" id="tgmpa-flavor-theme" name="tgmpa-flavor" value="parent-theme" required />
+				<label for="tgmpa-flavor-theme">Theme</label>
+				<br />
+				<input type="radio" id="tgmpa-flavor-child-theme" name="tgmpa-flavor" value="child-theme" required />
+				<label for="tgmpa-flavor-child-theme">Child Theme</label>
+				<br />
+				<input type="radio" id="tgmpa-flavor-plugin" name="tgmpa-flavor" value="plugin" required />
+				<label for="tgmpa-flavor-plugin">Plugin</label>
+			</fieldset>
+		</section>
+
+		<section class="tgmpa-form-secondary">
+			<fieldset>
+			<label for="tgmpa-text-domain">Text Domain:<span class="required">*</span></label>
+			<input type="text" id="tgmpa-text-domain" name="tgmpa-text-domain" placeholder="text-domain" value="" pattern="[A-Za-z0-9_-]+" title="Allowed characters: alphanumeric and dashes" autofocus required />
+			</fieldset>
+
+			<fieldset>
+			<label for="tgmpa-prefix">Function Prefix:</label>
+			<input type="text" id="tgmpa-prefix" name="tgmpa-prefix" placeholder="function_prefix" value="" pattern="[A-Za-z0-9_]+" title="Allowed characters: alphanumeric and dashes" />
+			</fieldset>
+
+			<fieldset>
+			<label for="tgmpa-name">Name:</label>
+			<input type="text" id="tgmpa-name" name="tgmpa-name" placeholder="Theme or Plugin Name" value="" pattern="[A-Za-z0-9 _-]+" title="Allowed characters: alphanumeric, space, underscores and dashes" />
+			</fieldset>
+
+			<fieldset id="tgmpa-form-wporg">
+				<input type="checkbox" id="tgmpa-wporg" name="tgmpa_wporg" value="1">
+				<label for="tgmpa-wporg">Do you intend to distribute the theme via wp.org ?</label>
```

Comment:
Oh dear, I though that battle had been fought (and won by Automattic). Will change.